### PR TITLE
build pybamm-base with no optional deps

### PR DESCRIPTION
**Pull request by Saransh-cpp**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by Saransh-cpp** [comment]
@conda-forge-admin, please rerender

**Comment by agriyakhetarpal** [comment]
Great idea, thanks for leading this effort, @Saransh-cpp! Please let me know if you would need help with testing this somehow. Would this approach remain possible with platform-specific recipes later on when we start on #65? Also, tagging @prady0t here from the PyBaMM issue in case he is interested in reviewing 👋

File: recipe/meta.yaml
Comment (agriyakhetarpal) [inline recipe/meta.yaml]: Didn't know this – we should try to package it. @prady0t, would you like to help out? We can get a few more maintainers from BPX itself added later on (it is the case for many feedstocks that maintainers don't fully overlap with the upstream project)

The docs for adding a new package are here: https://conda-forge.org/docs/maintainer/adding_pkgs/

Comment (Saransh-cpp) [reply recipe/meta.yaml]: Already on the way - https://github.com/conda-forge/staged-recipes/pull/28431 😉

**Comment by Saransh-cpp** [comment]
Tried debugging this for an hour but couldn't get it to run, and now that the hackathon is over, I would only be able to look at the error next week 😢

File: recipe/meta.yaml
Comment (Saransh-cpp) [inline recipe/meta.yaml]:
```suggestion
        - jax 0.4.27
        - jaxlib 0.4.27
```

File: recipe/meta.yaml
Comment (Saransh-cpp) [inline recipe/meta.yaml]:
```suggestion
```

File: recipe/meta.yaml
Comment (Saransh-cpp) [inline recipe/meta.yaml]:
```suggestion
        - bpx
```

File: recipe/meta.yaml
Comment (Saransh-cpp) [inline recipe/meta.yaml]:
```suggestion
```

File: recipe/meta.yaml
Comment (Saransh-cpp) [inline recipe/meta.yaml:122]: Everything except `dev`, `docs`, and `examples`.

File: recipe/meta.yaml
Comment (Saransh-cpp) [inline recipe/meta.yaml:43]: Required dependencies for `pybamm-base`.

**Review by Saransh-cpp — COMMENTED** [review]
Note for reviewers: please make sure that I did not miss any dependency 😬

File: recipe/meta.yaml
Comment (agriyakhetarpal) [inline recipe/meta.yaml:27]: I think we will also require `wheel` (note that `setuptools` injects `wheel` when building wheels) for `setuptools` <=70.1.0; we have to list it explicitly because we import it in our setup script to override the `bdist_wheel` class, so `setuptools`'s injection is not enough. It doesn't fail right now because the extension module isn't being built, but it is going to fail later. Once we bump to the noted `setuptools` version (I'll open a PR in PyBaMM and tag you there), we will be in a better position. Not something to do here, but we could note it as a comment here for future upgrades :)

File: recipe/meta.yaml
Comment (Saransh-cpp) [inline recipe/meta.yaml:62]:
```suggestion
        # See https://github.com/pybamm-team/PyBaMM/issues/1905
        # - pybamm.models.full_battery_models.lithium_metal
```

File: recipe/meta.yaml
Comment (Saransh-cpp) [inline recipe/meta.yaml:141]:
```suggestion
        # See https://github.com/pybamm-team/PyBaMM/issues/1905
        # - pybamm.models.full_battery_models.lithium_metal
```

**Review by agriyakhetarpal — APPROVED** [review]
Thanks, @Saransh-cpp! I wonder if we can document the fact that the "examples" extra isn't available in conda-forge's PyBaMM distribution and that users should run `conda/mamba install -c conda-forge jupyter` separately? Also, I think we should update the installation guide upstream about this change in general (and perhaps add an entry to the release notes for the next release as well).

File: recipe/meta.yaml
Comment (Saransh-cpp) [inline recipe/meta.yaml:27]:
```suggestion
        - setuptools >=64
        # See https://github.com/conda-forge/pybamm-feedstock/pull/70/files#r1876438836
        # - wheel
```

**Comment by Saransh-cpp** [comment]
Thanks for the review!

> document the fact that the "examples" extra isn't available in conda-forge's

Yes, I think this would require some work on upstream docs in general. I will open an issue in PyBaMM once this is merged.

> and perhaps add an entry to the release notes for the next release as well

Will mention this on slack.

**Pull request merged**
